tests/draft4/ref.json

```diff
@@ -174,6 +174,35 @@
         ]
     },
     {
+        "description": "nested ref with other properties",
+        "schema": {
+            "definitions": {
+                "a": {
+                    "type": "number"
+                }
+            },
+            "properties": {
+                "foo": {
+                    "type": "number",
+                    "maximum": 5,
+                    "$ref": "#/definitions/a"
+                }
```

Comment:
I would suggest not merging the test for $ref with other keywords in the same schema object, given the on-going discussion (@handrews) about whether they should be validated, ignored or prohibited (making the schema invalid).

I think the way we have in the spec now ("ignored") is the worst option, as it produces unexpected validation successes that are usually the most difficult to detect.

Probably, this test (and a similar one below) could be split into a separate PR so that everything else could be merged.